src/Configuration.cpp

```diff
@@ -191,15 +198,51 @@ void Configuration::parse (const std::string& input, int nest /* = 1 */)
         if (include != std::string::npos)
         {
           Path included (trim (line.substr (include + 7)));
-          if (included.is_absolute ())
+
+          do
           {
-            if (included.readable ())
-              load (included, nest + 1);
-            else
-              throw format ("Could not read include file '{1}'.", included._data);
+            // 0. Absolute path is not searched.
+            if (included.is_absolute ())
+              break;
+
+            // 1. Try relative to CWD first, break if exists. This is the legacy behavior.
+            if (included.exists ())
+              break;
+
+            // 2. Try path relative to the config file itself.
+            if (!file_path.empty ())
+            {
+              std::string file_dir = Path(file_path).realpath();
+              auto slash = file_dir.rfind('/');
+              file_dir.resize (slash != std::string::npos ? slash + 1 : 0); // `/` is kept.
+              Path file_relative (file_dir + included._data);
```

Comment:
I wonder if we should move this functionality to the `Path` object itself (finding parent dirs, concatenating paths). Perhaps a future improvement.